password-generator@spencerlommel.com/files/password-generator@spencerlommel.com/applet.js

```diff
@@ -51,17 +56,38 @@ class CinnamonRandomPasswordApplet extends Applet.IconApplet {
 
         this.copyButton.connect('clicked', () => {
             St.Clipboard.get_default().set_text(St.ClipboardType.CLIPBOARD, this.textBox.get_text());
+            if (this.playSound) {
+                this.playClickSound(); // Call play sound method after copying
+            }
         });
 
         this.onSettingsChanged(); // Apply initial settings
     }
 
     on_applet_clicked(event) {
-        this.menu.toggle();
         let newPassword = this.generateRandomPassword();
         this.textBox.set_text(newPassword);
+        if (this.autoCopy) {
+            St.Clipboard.get_default().set_text(St.ClipboardType.CLIPBOARD, newPassword);
+            this.playClickSound();
+        } else {
+            this.menu.toggle();
+        }
+    }
+
+    playClickSound() {
+        if (this.playSound && this.soundFilePath) {
+            let file = Gio.File.new_for_path(this.soundFilePath);
+            let player = new Gio.Subprocess({
+                argv: ['paplay', file.get_path()],
```

Comment:
Just a heads up that I believe this doesn't support .mp3 files. It's up to you if you want to implement this differently or to update the README so that users are aware.

Reply:
I attempted to use GSound and ran into some issues so I went with this route. My reasoning was that all of the linux mint system sounds are ogg sounds so that's what I wanted to primarily support

Reply:
Ah, that ties into the other comment. 😅